### PR TITLE
Trim empty rules.

### PR DESCRIPTION
**Pull request by wldcordeiro**
This PR introduces empty rule trimming, this isn't too useful to raw JSS (because you won't write empty rules), but with jss-nested you could end up with empty rules with a structure like:
```js
var style = {
  awesomeThing: {
    '& nestedAwesomeThing': {
      color: red;
    }
  }
};
```

### CODE REVIEW

**Comment by wldcordeiro** [comment]
I'm not sure whether this is a better solution or having each Rule type's `toString` return a null and check for the existence before joining. This way may only work with normal rules.

File: src/PluginsRegistry.js [hunk 1]
```diff
@@ -23,8 +23,8 @@ export default class PluginsRegistry {
    * @api private
    */
   run(rule) {
-    for (let index = 0; index < this.registry.length; index++) {
-      this.registry[index](rule)
-    }
+    this.registry.forEach(plugin => {
```
Comment (kof) [inline src/PluginsRegistry.js]: it was a micro performance optimization, because this runs on every rule.

File: src/StyleSheet.js [hunk 2]
```diff
@@ -129,8 +129,14 @@ export default class StyleSheet {
       if (stringified[rule.id]) {
         continue
       }
+
+      if (rule.style && Object.keys(rule.style).length === 0) {
```
Comment (kof) [inline src/StyleSheet.js]: As not every rule has a style property.

lets create a util isEmptyObject (separate file, same like clone)

http://jsperf.com/object-keys-length-vs-for-in-return
```
function isEmptyObject(obj) {
  for (var key in obj) return false
  return true
}
```

File: src/PluginsRegistry.js (hunk 1 above)
Comment (kof) [inline src/PluginsRegistry.js]: so please remove your change ... it generates a function on every .rule call.

File: src/empty-object.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export default function isEmptyObject(obj) {
```
Comment (kof) [inline src/empty-object.js]: file name should be is-empty-object

File: src/empty-object.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+  for (let key in obj) return false
```
Comment (kof) [inline src/empty-object.js]: should be const ...

Comment (kof) [inline src/empty-object.js]: didn't you get an error from linter?

File: src/StyleSheet.js [hunk 5]
```diff
@@ -129,8 +130,14 @@ export default class StyleSheet {
       if (stringified[rule.id]) {
         continue
       }
+
+      if (rule.style && isEmptyObject(rule.style)) {
```
Comment (kof) [inline src/StyleSheet.js]: empty check could be also part of isEmptyObject I think ...

File: src/empty-object.js (hunk 3 above)
Comment (kof) [inline src/empty-object.js]: would add here 
`if (!obj) return true`

File: src/StyleSheet.js (hunk 5 above)
Comment (kof) [inline src/StyleSheet.js]: please ignore this comment ...

File: src/empty-object.js (hunk 4 above)
Comment (wldcordeiro) [inline src/empty-object.js]: I did, but then it also errored about the const being unused, maybe update the eslint rules to allow `_` as an unused? That way the line can be `for (const _ in obj) return false`

File: src/empty-object.js (hunk 3 above)
Comment (kof) [inline src/empty-object.js]: pleae ignore that comment  ...

**Comment by kof** [comment]
merged, thanks!

**Pull request merged**